src/django/api/middleware.py

```diff
@@ -0,0 +1,45 @@
+import sys
+
+from django.conf import settings
+
+from rest_framework.authentication import get_authorization_header
+
+
+from api.models import RequestLog
+
+
+def _report_error_to_rollbar(request, auth):
+    ROLLBAR = getattr(settings, 'ROLLBAR', {})
+    if ROLLBAR:
+        import rollbar
+        rollbar.report_exc_info(
+            sys.exc_info(),
+            extra_data={'auth': auth})
+
+
+class RequestLogMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        auth = None
+        try:
+            if request.user and request.user.is_authenticated:
+                auth = get_authorization_header(request)
+                if auth and auth.split()[0].lower() == 'token'.encode():
```

Comment:
This code is based on
https://github.com/encode/django-rest-framework/blob/e4e75f1c7ceb053bffb8031e0468000c4d9c3cec/rest_framework/authentication.py#L170-L189